src/backend/distributed/commands/collation.c

```diff
@@ -519,6 +519,27 @@ DefineCollationStmtObjectAddress(Node *node, bool missing_ok)
 }
 
 
+/*
+ * PreprocessDefineCollationStmt executed before the collation has been
+ * created locally to ensure that the executing node is a coordinator
+ * if the created collation is set to propagate and the statement is not
+ * part of a multi statement transaction.
+ */
+List *
+PreprocessDefineCollationStmt(Node *node, const char *queryString,
+							  ProcessUtilityContext processUtilityContext)
+{
+	Assert(castNode(DefineStmt, node)->kind == OBJECT_COLLATION);
+
+	if (ShouldPropagate() && !IsMultiStatementTransaction())
```

Comment:
couldn't get the reason for that. Could you explain a bit? Or maybe it could be good to have some comments on the code

Reply:
Comment added, but because none of these lines were edited (new lines were added) girhub doesn't consider this portion of the code outdated.

Reply:
That will not call `EnsureCoordinator` if we are in a tx block. What I couldn't understand is, why do we allow the worker to execute `CREATE COLLATION` for non tx block cases?

Reply:
Well I can answer that by looking at the if statement:
* The worker will only be able to execute if the if statement evaluates to false or `EnsureCoordinator` is triggered and there is an error on a worker.
* For non transaction block cases, `!IsMultiStatementTransaction` is true, so the worker can only execute if `ShouldPropage` is false.

So the worker can only execute outside of a tx block when metadata syncing is off. This can happen when the user turns off metadata syncing manually, or when the CREATE statement is predicated from the Coordinator to the worker.

Reply:
sorry, my bad, forget my previous comment. The correct one should be:
"What I couldn't understand is, why do we allow the worker to execute CREATE COLLATION for ~non~ tx block cases?"

As far as I see, if there's an ongoing tx block, `CREATE COLLATION` will be executed on the worker?

Reply:
> "What I couldn't understand is, why do we allow the worker to execute CREATE COLLATION for non tx block cases?"
You are correct. I do not know why, but this behavior is tested for, so it is expected. @onderkalaci might have more insights regarding this.